tools/autograd/gen_python_functions.py

```diff
@@ -102,16 +102,15 @@ def should_generate_python_binding(declaration):
         if re.match('^' + pattern + '$', name):
             return False
 
+    # TODO: fix handling of SparseTensor. We don't want to generate Python
+    # bindings to SparseTensor overloads, such as add(Tensor, SparseTensor),
+    # since the Tensor-based signature already dynamically dispatches correctly.
+    # However, _sparse_mask only has a SparseTensor signature so we need to bind
+    # that function.
     for arg in declaration['arguments']:
-        if arg['type'] == 'SparseTensor':
+        if arg['type'] == 'SparseTensor' and declaration['name'] != '_sparse_mask':
```

Comment:
there are other cases where arg['type'] is SparseTensor?  Or arg['dynamic_type']?  (I always forget which is which.

Reply:
Not sure what you're asking. `arg['type'] == 'SparseTensor'` appears to be equivalent to `arg['dynamic_type'] == 'SparseTensor'`.

There are other functions which have overloads that take `SparseTensor`. They're:  add, add_, add_out. They get called dynamically from the non-SparseMask overload, so we don't want to bind them directly.

This is weird, but it's an artifact of the way we handle `SparseTensor` in ATen. It would probably be slightly better if `_sparse_mask` had the static type `Tensor` instead of `SparseTensor`